plugins/kubernetes-backend/src/service/KubernetesFetcher.ts

```diff
@@ -157,42 +188,49 @@ export class KubernetesClientBasedFetcher implements KubernetesFetcher {
       requestOptions.uri = requestOptions.uri.replace('/apis//v1/', '/api/v1/');
     });
 
-    if (namespace) {
-      return customObjects
-        .listNamespacedCustomObject(
-          resource.group,
-          resource.apiVersion,
-          namespace,
-          resource.plural,
-          '',
-          false,
-          '',
-          '',
-          labelSelector,
-        )
-        .then(r => {
-          return {
-            type: objectType,
-            resources: (r.body as any).items,
-          };
-        });
-    }
-    return customObjects
-      .listClusterCustomObject(
-        resource.group,
-        resource.apiVersion,
-        resource.plural,
-        '',
-        false,
-        '',
-        '',
-        labelSelector,
-      )
-      .then(r => {
-        return {
-          type: objectType,
-          resources: (r.body as any).items,
-        };
+    return this.namespaceExists(clusterDetails, namespace).then(exists => {
```

Comment:
If there is no namespace passed in we don't have to check for existence and can continue with the cluster listing logic. Wouldn't this logic would check empty namespaces existence unnecessarily?

Reply:
It will skip it.
```ts
// private namespaceExists(
// ...
    if (!namespace) {
      return Promise.resolve(false);
    }
````

Reply:
Can we move that check to the function so that is clear?

Reply:
Done